### PR TITLE
Add HTTPS status checks

### PR DESCRIPTION
**Pull request by JanLauGe**
Hello!
**TLDR; I think the package should provide verbose error messages for HTTPS requests. This PR is a suggestion. It will need more work but I want to check with you before spending more time on it.**

I started using the package and noticed that I get a lot of cryptic errors when playing around with the API. With a little bit of debugging I found that I was sending invalid requests to the Robinhood API. However, the error messages I was getting back were cryptic `dplyr` errors. Looking at the issues in the forum, I think that there are a couple of users who have similar problems but don't understand what's happening. I think the situation could be improved by adding a simple check to any `GET` request for status `200`. If it is anything else, a warning or error message should be printed to aid the user in trouble shooting.

Suggested change in the PR is a little helper function that I would basically add in after every `GET` request.

Let me know what you think!

### CODE REVIEW

**Comment by JestonBlu** [comment]
Hi @JanLauGe. Thanks for the the PR and the idea, I like it. I like the idea of addressing it through a function. Im guessing this is what you are thinking, but in your commit in the api function you have the if/then logic where I would think you would have a reference to the function you have created. all of the api functions more or less follow the same format so you might be able to implement that function pretty easily. Is this what you are thinking?

**Comment by JanLauGe** [comment]
You are right! I started with the `if else` logic and then realized there is lots of `GET` calls in different places, so switched to a function. Instead of the `if else` bit we would call the function after every `GET` call.

After thinking about it some more though, I actually think the best way to do it would be to use the `httr::` functions. These are easy to use, have very descriptive names, good status checking and error handling out of the box. I see you already import `httr::` so I will amend the PR to use those functions unless you tell me you would prefer the original suggestion.

Example:

```
# GET call
dta <- GET(url,
           add_headers("Accept" = "application/json",
                       "Content-Type" = "application/json",
                       "Authorization" = token))

warn_for_status(dta)
# Or?
stop_for_status(dta)
```
see: [httr status checks](https://cran.r-project.org/web/packages/httr/vignettes/quickstart.html#cb5-2)

**Comment by JestonBlu** [comment]
I think your new suggestion is better. Good to take advantage of httr I think. I think 'stop_for_status' would be better because if it fails at that point it's the downstream stuff that is returning the cryptic errors. Better to just cut it off there.

**Comment by JanLauGe** [comment]
Thanks!

That's done but something weird is going on with my pull request. Lots of lines highlighted as changed but I don't see a change. Do you know what's up?

**Comment by JestonBlu** [comment]
Yeah i see what you mean. Im not really sure.

**Comment by JestonBlu** [comment]
Thanks for the contribution. Feel free to submit more!

**Comment by JestonBlu** [comment]
@JanLauGe I found one issue, sorry, your functions are referring to `df` instead of `dta`. If you fix that and resubmit it will compile correctly and then its good.
